documentation/modules/ROOT/pages/connectors/postgresql.adoc

```diff
@@ -2256,6 +2256,94 @@ Again, regularly emitting events solves the problem.
+
+The following procedure lists the steps for performing a PostgreSQL database upgrade so that {prodname} continues to capture events, and no data loss occurs:
+
+.Procedure
+
+1. Ensure that no data changes can occur during the upgrade process by temporarily stopping applications that write to the database, or putting them into a read-only mode.
+
+2. Back up the database.
+
+3. Temporarily disable write access to the database.
+
+4. Verify that any changes that occured in the database before you blocked write operations are saved to the WAL on the replication slot.
+
+5. Provide the connector with enough time to capture all event records that are written to the replication slot. +
+This step ensures that all change events that occured before the downtime are accounted for, and that they are saved to Kafka.
+
+6. Verify that the connector has finished consuming entries from the replication slot by checking the value of the flushed LSN.
+
+7. Shut down the connector gracefully by stopping Kafka Connect. +
+Kafka Connect stops the connectors, flushes all event records to Kafka, and records the last offset received from each connector.
+//   Do we need to delete the connector and its offset topic?
+
+8. As a PostgreSQL administrator, drop the replication slot on the primary database server.
+//   Can this be done via setting xref:postgresql-property-slot-drop-on-stop[`slot.drop.on.stop`] to `true`?
+
+9. Stop the database.
+
+10. Perform the upgrade using an approved PostgreSQL upgrade procedure, such as `pg_upgrade` or `pg_dump/restore`.
+
+11. (Optional) Use a standard Kafka tool to remove the connector offsets from the offset storage topic. +
+For an example of how to remove connector offsets, see https://debezium.io/documentation/faq/#how_to_remove_committed_offsets_for_a_connector[how to remove connector offsets] in the {prodname} community FAQ.
+
+12. Restart the database.
+
+13. As a PostgreSQL administrator, create a {prodname} logical replication slot on the database.
+You must create the slot before enabling writes to the database.
+Otherwise, {prodname} cannot capture the changes, resulting in data loss.
+
+14. As a PostgreSQL administrator, create a publication that defines the tables to be captured.
+//   Can this step be skipped, assuming that a publication was previously configured, and it's preserved during upgrade?
+ifdef::product[]
++
+For information about setting up a replication slot, see xref:configuring-a-replication-slot-for-the-debezium-pgoutput-plug-in[].
+endif::product[]
+
+15. In the {prodname} connector configuration, set the xref:postgresql-property-publication-name[`publication.name`] property to the name of the publication.
+//   Is this necessary? Is the previous configured publication name still valid? Should automatic creation be disabled? (i.e., `publication.autocreate.mode` set to `disabled`)]
```

Comment:
I'd modfiy this a bit saying something like if publication is not present after database upgrade it should be re-created. This is more like an insurance step as IIRC the publiations should survive the upgrade.